### ISSUE TITLE
Adding external references is failing when the user has a lower confidence level than the entity

### ISSUE THREAD

**Issue by gileri** [LuatixHQ/opencti#9171]
## Description

[The sentinel-intel connector adds a relationship after publishing data to the Sentinel API](https://github.com/OpenCTI-Platform/connectors/blob/fbdab76582dff61262068d17687292d20861c6fc/stream/sentinel-intel/src/sentinel_intel_connector/connector.py#L115-L131
).

However, it fails when it does so on a entity with an higher Confidence Level than the maximum Confidence Level set on the connector user.

## Environment

1. OS (where OpenCTI server runs): Official docker image
2. OpenCTI version: 6.4.0
3. OpenCTI client: python
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Run the connector
2. Create an indicator that is picked up by the connector stream

## Expected Output

No error, external reference is created

## Actual Output

```json
{"timestamp": "2024-11-26T11:35:04.191941Z", "level": "ERROR", "name": "Sentinel Intel",
[REDACTED]
"exc_info": "Traceback (most recent call last):\n  File \"/opt/opencti-connector-sentinel-intel/sentinel_intel_connector/connector.py\", line 256, in process_message\n    self._handle_create_event(data)\n  File \"/opt/opencti-connector-sentinel-intel/sentinel_intel_connector/connector.py\", line 188, in _handle_create_event\n    self._create_sentinel_indicator(observable)\n  File \"/opt/opencti-connector-sentinel-intel/sentinel_intel_connector/connector.py\", line 123, in _create_sentinel_indicator\n    self.helper.api.stix_domain_object.add_external_reference(\n  File \"/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_stix_domain_object.py\", line 1703, in add_external_reference\n    self.opencti.query(\n  File \"/usr/local/lib/python3.11/site-packages/pycti/api/opencti_api_client.py\", line 368, in query\n    raise ValueError(value_error)\nValueError: {'name': 'FUNCTIONAL_ERROR', 'error_message': 'User effective max confidence level is insufficient to update this element'}"}
```

**Comment by SamuelHassine** [comment]
Transferring to OpenCTI as adding an external reference to an entity with a lower confidence level should be possible. Also, I'm seeing a risk here to have the external reference created but not associated to anything.

**Comment by nino-filigran** [comment]
Just to be clear @SamuelHassine : here the issue is that it fails to attach an external reference to an entity that has **higher** confidence level than the confidence level of the connector, and not **lower** as you have mentionned. Am I missing anything? And so far, this behavior is the wanted behavior in the platform, given that we don't want users with lower confidence to edit entities with higher confidence.

**Comment by romain-filigran** [comment]
We need to make sure that it makes sense to update the indicators pushed in Sentinel by adding an external reference. In my opinion, a “stream connector” should only export indicators, not update them. 
Is there really any advantage to adding these external references and adding a link to the Sentinel indicator identifier?
@SamuelHassine : any opinion ?

**Comment by SamuelHassine** [comment]
@romain-filigran @nino-filigran, we should have discussed this long time ago. This issue is not about connectors. I agree with you on the principle that a stream connector should not update an entity directly through API, but here, we were talking about a new issue on the fact that when the confidence level of a user is below the existing entity, the user cannot add an external reference.

**This is a massive regression in the overall approach of the platform, to consolidate data**. This should be fixed asap.

I've multiple connectors crashing, including the one I'm testing now:

```
{"timestamp": "2025-03-08T20:12:07.330213Z", "level": "ERROR", "name": "Tanium Threat Response Intel", "message": "Error in ListenStream loop, exit.", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.12/site-packages/pycti/connector/opencti_connector_helper.py\", line 658, in run\n    self.callback(msg)\n  File \"/opt/opencti-connector-tanium-intel/tanium_intel_connector/connector.py\", line 125, in process_message\n    self._handle_create_event(data)\n  File \"/opt/opencti-connector-tanium-intel/tanium_intel_connector/connector.py\", line 72, in _handle_create_event\n    self.intel_manager.create_intel_from_indicator(data)\n  File \"/opt/opencti-connector-tanium-intel/tanium_intel_connector/intel_manager.py\", line 113, in create_intel_from_indicator\n    self._add_entity_external_reference(indicator, intel_document[\"id\"])\n  File \"/opt/opencti-connector-tanium-intel/tanium_intel_connector/intel_manager.py\", line 36, in _add_entity_external_reference\n    self.helper.api.stix_domain_object.add_external_reference(\n  File \"/usr/local/lib/python3.12/site-packages/pycti/entities/opencti_stix_domain_object.py\", line 1701, in add_external_reference\n    self.opencti.query(\n  File \"/usr/local/lib/python3.12/site-packages/pycti/api/opencti_api_client.py\", line 371, in query\n    raise ValueError(value_error)\nValueError: {'name': 'FUNCTIONAL_ERROR', 'error_message': 'User effective max confidence level is insufficient to update this element'}", "taskName": null, "attributes": {"reason": "{'name': 'FUNCTIONAL_ERROR', 'error_message': 'User effective max confidence level is insufficient to update this element'}"}}
```

This is not the expected behavior, Tanium connector should be able to add the external reference regardless the confidence level. Swithing this issue as critical as we are missing so much contextual data in all external import connectors... Really not expected as we are consolidating external references and labels since version 1.0.

**To be clear: adding an external reference or a label is authorized when having a lower confidence level, updating existing fields is not authorized.**

**Comment by SamuelHassine** [comment]
> Just to be clear [@SamuelHassine](https://github.com/SamuelHassine) : here the issue is that it fails to attach an external reference to an entity that has **higher** confidence level than the confidence level of the connector, and not **lower** as you have mentionned. Am I missing anything? And so far, this behavior is the wanted behavior in the platform, given that we don't want users with lower confidence to edit entities with higher confidence.

No, this was not the expected behavior. Editing yes, adding external references and/or labels no.

**Comment by SamuelHassine** [comment]
> We need to make sure that it makes sense to update the indicators pushed in Sentinel by adding an external reference. In my opinion, a “stream connector” should only export indicators, not update them. Is there really any advantage to adding these external references and adding a link to the Sentinel indicator identifier? [@SamuelHassine](https://github.com/SamuelHassine) : any opinion ?

It is always valuable to add references pointing the external resources we have, whether it is incidents, indicators etc. to allow people to easly pivot between tools.
